### ISSUE TITLE
memory allocation related to prepare_queue_max_size needs to be reviewed

### ISSUE THREAD

**Issue by rao-vasireddy** [YugaByte/yugabyte-db#511]
InitTabletPeer calls:

```
prepare_thread_ = std::make_unique<Preparer>(consensus_.get(), tablet_prepare_pool);
```
And
```
PreparerImpl::PreparerImpl(consensus::Consensus* consensus,
ThreadPool* tablet_prepare_pool)
: consensus_(consensus),
queue_(FLAGS_prepare_queue_max_size),
tablet_prepare_pool_token_(tablet_prepare_pool
->NewToken(ThreadPool::ExecutionMode::SERIAL)) {
}
```

There appears to be a high fixed overhead per tablet due to the prepare pool's lockless queue... 
```
DEFINE_int32(prepare_queue_max_size, 100000,
"Maximum number of operations waiting in the per-tablet prepare queue.");
/ Note that the lock-free queue seems to be preallocating memory proportional to the queue size
// (about 64 bytes per entry for 8-byte pointer keys) – something to keep in mind with a large
// number of tablets.
DEFINE_int32(prepare_queue_max_size, 100000,
"Maximum number of operations waiting in the per-tablet prepare queue.");
```

And on low memory machines this can be a problem.

**Comment by rao-vasireddy** [comment]
[pprof_alloc_bytes.docx](https://github.com/YugaByte/yugabyte-db/files/2508009/pprof_alloc_bytes.docx)

**Comment by kmuthukk** [comment]
[pprof.alloc_bytes.html.txt](https://github.com/YugaByte/yugabyte-db/files/2508019/pprof.alloc_bytes.html.txt)

**Comment by rao-vasireddy** [comment]
[pprof_alloc_bytes.pdf](https://github.com/YugaByte/yugabyte-db/files/2508082/pprof_alloc_bytes.pdf)

**Comment by spolitov** [comment]
Fixed by b733351def7756d842a1007933bc334b3d1d6a18
